### PR TITLE
vdpa_sim_utils.py: add module for VDPA net device interface.

### PR DESCRIPTION
**Pull request by quanwenli**
vdpa_sim_utils.py: add module for VDPA net device interface

    vdpa_sim_utils.py:
      1. repalce filename from vdpa_sim_blk_utils to vdpa_sim_utils
      2. update the VDPA block interfaces
      3. add Class VhostVdpaNetSimulator
    block_vhost_vdpa_test.py: point to new VhostVdpaBlkSimulatorTest
    vdpa_sim_blk_test.py: point to new VhostVdpaBlkSimulatorTest

ID:1929
Signed-off-by: Wenli Quan <wquan@redhat.com>

### CODE REVIEW

**Comment by quanwenli** [comment]
@qingwangrh could you help review this ?  thanks

**Comment by qingwangrh** [comment]
reference test cmd, @quanwenli , please post your test result
python ConfigTest.py --testcase=block_vhost_vdpa_test,vdpa_sim_blk_test --guestname=RHEL.9.4.0 --firmware=default_bios --clone=no

File: provider/vdpa_sim_utils.py [hunk 1]
```diff
@@ -44,45 +44,46 @@ def unload_modules(self):
...
 
 
-class VhostVdpaSimulatorTest(VDPABlkSimulatorTest):
+class VhostVdpaBlkSimulatorTest(VDPABlkNetSimulatorTest):
     def __init__(self):
-        super(VhostVdpaSimulatorTest, self).__init__()
+        super(VhostVdpaBlkSimulatorTest, self).__init__()
         self._modules = ['vhost-vdpa', 'vdpa-sim-blk']
 
     def add_vdpa_blk_dev(self, name):
```
Comment (qingwangrh) [inline provider/vdpa_sim_utils.py:89]: change to add_dev

File: provider/vdpa_sim_utils.py [hunk 2]
```diff
@@ -44,45 +44,46 @@ def unload_modules(self):
...
-    def add_vdpa_blk_dev(self, name):
+    def add_dev(self, name):
         """
-        Add vDPA blk device
+        Add vDPA device
         :param name: device name
         :return : host device name ,eg. vda,vdb
         """
-        raise VDPABlkSimulatorError("Please implement add_vdpa_blk_dev")
+        raise VDPABlkNetSimulatorError("Please implement add_vdpa_blk_dev"
```
[the remark this answers is not in the document]
Comment (quanwenli) [reply provider/vdpa_sim_utils.py]: updated.

File: provider/vdpa_sim_utils.py (hunk 1 above)
Comment (quanwenli) [reply provider/vdpa_sim_utils.py]: udpated.

File: provider/vdpa_sim_utils.py [hunk 3]
```diff
@@ -101,14 +102,14 @@ def add_vdpa_blk_dev(self, name):
...
 
 
-class VirtioVdpaSimulatorTest(VDPABlkSimulatorTest):
+class VirtioVdpaBlkSimulatorTest(VDPABlkNetSimulatorTest):
     def __init__(self):
-        super(VirtioVdpaSimulatorTest, self).__init__()
+        super(VirtioVdpaBlkSimulatorTest, self).__init__()
         self._modules = ['virtio-vdpa', 'vdpa-sim-blk']
 
     def add_vdpa_blk_dev(self, name):
```
[the remark this answers is not in the document]
Comment (quanwenli) [reply provider/vdpa_sim_utils.py]: updated

File: provider/vdpa_sim_utils.py [hunk 4]
```diff
@@ -130,5 +131,32 @@ def add_vdpa_blk_dev(self, name):
...
+            raise VDPABlkNetSimulatorError("vdpa dev add %s failed" % name)
         return host_dev[0]
+
+
+class VhostVdpaNetSimulatorTest(VDPABlkNetSimulatorTest):
+    def __init__(self):
+        super(VhostVdpaNetSimulatorTest, self).__init__()
+        self._modules = ['vhost-vdpa', 'vdpa-sim', 'vdpa-sim-net']
+
+    def add_vdpa_net_dev(self, name, mac):
```
[the remark this answers is not in the document]
Comment (quanwenli) [reply provider/vdpa_sim_utils.py]: udpated

**Comment by quanwenli** [comment]
(1/8) Host_RHEL.m9.u4.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.vdpa_sim_blk_test.multi_disks.q35: STARTED
 (1/8) Host_RHEL.m9.u4.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.vdpa_sim_blk_test.multi_disks.q35: PASS (101.12 s)
 (2/8) Host_RHEL.m9.u4.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.vdpa_sim_blk_test.hotplug_unplug.q35: STARTED
 (2/8) Host_RHEL.m9.u4.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.vdpa_sim_blk_test.hotplug_unplug.q35: PASS (115.85 s)
 (3/8) Host_RHEL.m9.u4.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.vdpa_sim_blk_test.discard_check.q35: STARTED
 (3/8) Host_RHEL.m9.u4.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.vdpa_sim_blk_test.discard_check.q35: PASS (8.05 s)
 (4/8) Host_RHEL.m9.u4.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_vhost_vdpa_test.multi_disks.q35: STARTED
 (4/8) Host_RHEL.m9.u4.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_vhost_vdpa_test.multi_disks.q35: PASS (102.42 s)
 (5/8) Host_RHEL.m9.u4.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_vhost_vdpa_test.hotplug_unplug.q35: STARTED
 (5/8) Host_RHEL.m9.u4.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_vhost_vdpa_test.hotplug_unplug.q35: PASS (117.85 s)
 (6/8) Host_RHEL.m9.u4.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_vhost_vdpa_test.with_discard.q35: STARTED
 (6/8) Host_RHEL.m9.u4.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_vhost_vdpa_test.with_discard.q35: PASS (102.42 s)
 (7/8) Host_RHEL.m9.u4.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_vhost_vdpa_test.with_cache.with_direct_on.q35: STARTED
 (7/8) Host_RHEL.m9.u4.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_vhost_vdpa_test.with_cache.with_direct_on.q35: PASS (102.52 s)
 (8/8) Host_RHEL.m9.u4.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_vhost_vdpa_test.with_cache.with_direct_off.q35: STARTED
 (8/8) Host_RHEL.m9.u4.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_vhost_vdpa_test.with_cache.with_direct_off.q35: PASS (11.62 s)

**Comment by quanwenli** [comment]
@XueqiangWei could you help review this ? thanks.

**Comment by qingwangrh** [comment]
LGME

**Comment by XueqiangWei** [comment]
I think we need to add separate commit to the three files. like autotest/tp-qemu/pull/3836
e.g.

vdpa_sim_utils.py
vdpa_sim_utils: add module for VDPA net device interface
1. change the filename from vdpa_sim_blk_utils to vdpa_sim_utils
2. update the VDPA block interfaces and add Class VhostVdpaNetSimulatorTest

block_vhost_vdpa_test.py
vdpa_sim_blk_test.py
VDPA block device test: point to new function
1. block_vhost_vdpa_test: point to new VhostVdpaBlkSimulatorTest
2. vdpa_sim_blk_test: point to new VhostVdpaBlkSimulatorTest

**Comment by XueqiangWei** [comment]
@quanwenli Do you have a new automated testing case to validate the new Class VhostVdpaNetSimulatorTest?

**Comment by quanwenli** [comment]
> @quanwenli Do you have a new automated testing case to validate the new Class VhostVdpaNetSimulatorTest?

I plan to have, and need this patch merged firstly.

**Comment by quanwenli** [comment]
> 2\. add Class VhostVdpaNetSimulatorTest

I have update the commit. please help review.

**Comment by XueqiangWei** [comment]
I mean add separate commit to the three files.  Please refer to autotest/tp-qemu/pull/3836/commits

**Comment by quanwenli** [comment]
> I mean add separate commit to the three files. Please refer to [/pull/3836](https://github.com/autotest/tp-qemu/pull/3836)/commits

I have updated. please help reveiw.

**Comment by XueqiangWei** [comment]
(01/11) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86
_64.io-github-autotest-qemu.vdpa_sim_blk_test.multi_disks.q35: STARTED
 (01/11) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86
_64.io-github-autotest-qemu.vdpa_sim_blk_test.multi_disks.q35: PASS (73.60 s)
 (02/11) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86
_64.io-github-autotest-qemu.vdpa_sim_blk_test.hotplug_unplug.q35: STARTED
 (02/11) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86
_64.io-github-autotest-qemu.vdpa_sim_blk_test.hotplug_unplug.q35: PASS (126.04 s)
 (03/11) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86
_64.io-github-autotest-qemu.vdpa_sim_blk_test.discard_check.q35: STARTED
 (03/11) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86
_64.io-github-autotest-qemu.vdpa_sim_blk_test.discard_check.q35: PASS (6.81 s)
 (04/11) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86
_64.io-github-autotest-qemu.block_vhost_vdpa_test.multi_disks.q35: STARTED
 (04/11) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86
_64.io-github-autotest-qemu.block_vhost_vdpa_test.multi_disks.q35: PASS (59.26 s)
 (05/11) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_vhost_vdpa_test.hotplug_unplug.q35: STARTED
 (05/11) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_vhost_vdpa_test.hotplug_unplug.q35: PASS (74.26 s)
 (06/11) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_vhost_vdpa_test.with_discard.q35: STARTED
 (06/11) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_vhost_vdpa_test.with_discard.q35: PASS (113.88 s)
 (07/11) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_vhost_vdpa_test.with_cache.with_direct_on.q35: STARTED
 (07/11) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_vhost_vdpa_test.with_cache.with_direct_on.q35: PASS (114.37 s)
 (08/11) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_vhost_vdpa_test.with_cache.with_direct_off.q35: STARTED
(08/11) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_vhost_vdpa_test.with_cache.with_direct_off.q35: PASS (14.26 s)
 (09/11) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_vhost_vdpa_test.with_detect_zeroes.with_discard_unmap.q35: STARTED
 (09/11) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_vhost_vdpa_test.with_detect_zeroes.with_discard_unmap.q35: PASS (114.39 s) 
 (10/11) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_vhost_vdpa_test.with_detect_zeroes.with_discard_ignore.q35: STARTED
 (10/11) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_vhost_vdpa_test.with_detect_zeroes.with_discard_ignore.q35: PASS (114.13 s)
 (11/11) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_vhost_vdpa_test.with_read_only.q35: STARTED
 (11/11) Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.block_vhost_vdpa_test.with_read_only.q35: PASS (113.65s)

**Review by XueqiangWei — APPROVED** [review]
LGTM,
Acked-by: xuwei@redhat.com

**Pull request merged**